### PR TITLE
Enable saving a cell to a dashboard

### PR DESCRIPTION
**Pull request by chnn**
Enable a user to click the _Add New Cell_ button to open the VEO. It will be populated with a default line view (that has a hardcoded query for now). The view can be saved to a dashboard in a cell.

If a user edits a cell, the VEO will be populated with all the state from the cell's view.

### CODE REVIEW

File: ui/src/dashboards/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -265,7 +270,10 @@ class DashboardPage extends Component<Props, State> {
   }
 
   private handleAddCell = async (): Promise<void> => {
-    const newView = getNewView()
+    // TODO: We should revisit our polymorphic `View` type
```
Comment (chnn) [inline ui/src/dashboards/containers/DashboardPage.tsx]: I'm going to look into this next. Our typing of the various view types is pretty painful at the moment.

File: ui/src/dashboards/components/VEO.tsx [hunk 2]
```diff
@@ -21,12 +18,12 @@ import {View} from 'src/types/v2/dashboards'
 import {TimeMachineTab} from 'src/types/v2/timeMachine'
 
 interface StateProps {
-  name: string
+  draftView: View
```
Comment (alexpaxton) [inline ui/src/dashboards/components/VEO.tsx]: I like this `draft` naming convention, easy to understand

File: ui/tslint.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
     ],
     "no-empty": false,
     "no-empty-interface": false,
+    "no-shadowed-variable": false,
```
Comment (alexpaxton) [inline ui/tslint.json]: Yay

Comment (chnn) [reply ui/tslint.json]: :)

**Pull request merged**